sss.go

```diff
@@ -11,19 +11,93 @@ import (
 	"github.com/lestrrat-go/jwx/jwe"
 )
 
-// DecryptSss implements Shamir Secret Sharing decryption algorithm
-func DecryptSss(msg *jwe.Message, clevisNode map[string]interface{}) ([]byte, error) {
-	sssNode, ok := clevisNode["sss"].(map[string]interface{})
-	if !ok {
-		return nil, fmt.Errorf("clevis.go/sss: cannot parse provided token, node 'clevis.sss'")
+// SssPin represents the data samir secret sharing needs to perform decryption
+type SssPin struct {
+	Prime     string   `json:"p"`
+	Threshold int      `json:"t"`
+	Jwe       []string `json:"jwe"`
+}
+
+func appendConfig(slice []json.RawMessage, obj interface{}) ([]json.RawMessage, error) {
+	cfgBytes, err := json.Marshal(obj)
+	if err != nil {
+		return slice, err
 	}
+	return append(slice, cfgBytes), nil
+}
 
-	primeNode, ok := sssNode["p"].(string)
-	if !ok {
-		return nil, fmt.Errorf("clevis.go/sss: cannot parse provided token, node 'clevis.sss.p'")
+// ToConfig converts a given SssPin into the corresponding SssConfig which can be used for encryption
+func (p SssPin) ToConfig() (SssConfig, error) {
+	c := SssConfig{
+		Threshold: p.Threshold,
+		Pins:      make(map[string][]json.RawMessage),
 	}
+	jweErrors := make([]error, 0)
+	for _, jwe := range p.Jwe {
+		_, pin, err := Parse([]byte(jwe))
+		if err != nil {
+			jweErrors = append(jweErrors, err)
```

Comment:
This kind of error accumulation look not very Go idiomatic. Would it make a sense return the first error and use `return err` everywhere in this function?

Reply:
Fixed with a more go-like fail-first approach.